blueprints/data-solutions/dataproc-lifecycle-via-composer/README.md

```diff
@@ -0,0 +1,115 @@
+}
+```
+4. (Optional) You can run `python3 ../include/generate_dag_files.py` in your local environment, if you want to review generated DAGs before deploying(TF plan/apply) those.
+
+## Deployment
+
+1. Set Google Cloud Platform credentials on local environment: 
+https://cloud.google.com/source-repositories/docs/authentication
+
+
+2. You must supply the following terraform variables in order to deploy the project:
+   - `project_id`: Project ID where all the resources will be created, including; Cloud Composer environment, Dataproc cluster, policies, buckets, etc.
+   - `region`: Region where all the resources will be created, including; Cloud Composer environment, Dataproc cluster, policies, buckets, etc.
+   - `network_config`: The example supports the configuration of a Shared VPC as an input variable. To deploy the solution on a Shared VPC, you have to configure the network_config variable. Includes:
+     - `host_project`: The Shared VPC host project ID
+     - `network_self_link`: The Shared VPC network URI
+     - `subnet_self_link`: The subnet in the Shared VPC subnetwork URI
+   - `dataproc_config`: includes:
+     - `composer_env_name`: Name of Cloud Composer environment to be created.
+     - `autoscaling_policy_id`: Dataproc Cluster autoscaling policy Id to be created and reused by Dataproc clusters.
+   
+
+3. Run Terraform Plan/Apply
+   ```bash
+    $ cd terraform/
+    $ terraform init
+    $ terraform plan
+    $ terraform apply
+   ##Optionally variables could be used
+    $ terraform apply -var 'project_id=<PROJECT_ID>' \
+   -var 'region=<REGION>'
+    ```
+Default Terraform variables and example values in `varibles.tf` file.
+
+Once you deploy terraform plan for the first time, and composer environment is running, you can _terraform plan/apply_ after adding new DAG configuration files, to generate and deploy DAGs, to the existing environment. 
+
+First time it is deployed, resource creation will take a few minutes because of Composer Environment provisioning, you should expect for successful completion along with a list of the created resources.
+
+## Running DAGs
+DAGs will run as per **Schedule**, **StartDate**, and **Catchup** configuration in DAG configuration file, or can be triggered manually trough Airflow web console after deployment. 
+![](images/dag_execution_example.png)
+
+## Shared VPC
+The example supports the configuration of a Shared VPC as an input variable. 
+To deploy the solution on a Shared VPC, you have to configure the `network_config` variable:
+
+```
+network_config = {
+    host_project       = "PROJECT_ID"
+    network_self_link  = "https://www.googleapis.com/compute/v1/projects/PROJECT_ID/global/networks/VPC_NAME"
+    subnet_self_link   = "https://www.googleapis.com/compute/v1/projects/PROJECT_ID/regions/$REGION/subnetworks/SUBNET_NAME"
+  }
+```
```

Comment:
I would mention that firewall rules would be needed in the host project, point to the documentation: 

https://cloud.google.com/composer/docs/how-to/managing/configuring-private-ip#step_3_configure_firewall_rules

Reply:
Added